lib/compiler-cranelift/src/config.rs

```diff
@@ -117,11 +117,13 @@ impl Cranelift {
             builder.enable("has_lzcnt").expect("should be valid flag");
         }
 
-        builder.finish(self.flags())
+        let is_riscv = matches!(target.triple().architecture, Architecture::Riscv64(_));
+
+        builder.finish(self.flags(is_riscv))
     }
 
     /// Generates the flags for the compiler
-    pub fn flags(&self) -> settings::Flags {
+    pub fn flags(&self, is_riscv: bool) -> settings::Flags {
```

Comment:
Since this is a public function, I think it will be better to pass the `&Target` to this function (similarly to isa)

```suggestion
    pub fn flags(&self, target: &Target) -> settings::Flags {
        let is_riscv = matches!(target.triple().architecture, Architecture::Riscv64(_));
```

Reply:
Ok, done (not sure why `flags` is public)